test/e2e/memcached_test.go

```diff
@@ -582,8 +594,98 @@ func MemcachedClusterTest(t *testing.T) {
 	cmdOut, err := exec.Command("operator-sdk", "test", "cluster", *e2eImageName,
 		"--namespace", namespace,
 		"--image-pull-policy", "Never",
-		"--service-account", "memcached-operator").CombinedOutput()
+		"--service-account", operatorName).CombinedOutput()
 	if err != nil {
 		t.Fatalf("In-cluster test failed: %v\nCommand Output:\n%s", err, string(cmdOut))
 	}
 }
+
+func memcachedMetricsTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+
+	// Make sure metrics Service exists
+	s := v1.Service{}
+	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: operatorName, Namespace: namespace}, &s)
+	if err != nil {
+		return fmt.Errorf("could not get metrics Service: (%v)", err)
+	}
+
+	// Get operator pod
+	pods := v1.PodList{}
+	opts := client.InNamespace(namespace)
+	if len(s.Spec.Selector) == 0 {
+		return fmt.Errorf("no labels found in metrics Service")
+	}
+
+	for k, v := range s.Spec.Selector {
+		if err := opts.SetLabelSelector(fmt.Sprintf("%s=%s", k, v)); err != nil {
+			return fmt.Errorf("failed to set list label selector: (%v)", err)
+		}
+	}
+
+	if err := opts.SetFieldSelector("status.phase=Running"); err != nil {
+		return fmt.Errorf("failed to set list field selector: (%v)", err)
+	}
+	err = f.Client.List(context.TODO(), opts, &pods)
+	if err != nil {
+		return fmt.Errorf("failed to get pods: (%v)", err)
+	}
+	podName := ""
+	if len(pods.Items) > 1 {
+		// If we got more than one pod, get leader pod name.
+		leader, err := verifyLeader(t, namespace, f, s.Spec.Selector)
+		if err != nil {
+			return err
+		}
+		podName = leader.Name
+		if podName == "" {
+			return fmt.Errorf("failed to get leader pod name")
+		}
+	}
+	if podName == "" {
+		podName = pods.Items[0].Name
+	}
```

Comment:
If `len(pods.Items)` is zero, will this section of code panic? I'd suggest refactoring to something like the below:

```go
podName := ""
if len(pods.Items) == 1 {
    podName = pods.Items[0].Name
} else if len(pods.Items) > 1 {
    // get leader pod name
} else {
    return fmt.Errorf("failed to get operator pod: service did not select any pods")
}

Reply:
> will this section of code panic

Yes, I tested this, and it does indeed panic, nice catch! My assumption that `.List` will just error out was wrong. Done, also added a todo comment to remove this logic when we get metrics in all pods, not just leader.